web/src/peer_data.ts

```diff
@@ -29,6 +71,16 @@ function get_user_set(stream_id: number): LazySet {
     return subscribers;
 }
 
+async function get_full_user_set(stream_id: number): Promise<LazySet> {
+    // This function parallels `get_user_set` but ensures we include all
+    // subscribers, possibly fetching that data from the server.
+    if (!fetched_stream_ids.has(stream_id) && !page_params.is_spectator) {
```

Comment:
I think you want to check `page_params.is_spectator` and return early? That seems clearer.

Reply:
`assert` seems right, and I tried clicking around in spectator view and had no issues with that.